46sshd/module-setup.sh

```diff
@@ -62,8 +64,8 @@ install() {
     inst_simple "${moddir}/sshd.service" "$systemdsystemunitdir/sshd.service"
     inst_simple "${moddir}/sshd_config" /etc/ssh/sshd_config
 
-    grep '^sshd:' /etc/passwd >> "$initdir/etc/passwd"
-    grep '^sshd:' /etc/group  >> "$initdir/etc/group"
+    (grep '^sshd:' /etc/passwd || getent passwd sshd || echo 'sshd:x:74:74:Privilege-separated SSH:/var/empty/sshd:/sbin/nologin' ) >> "$initdir/etc/passwd"
+    (grep '^sshd:' /etc/group || getent group sshd || echo 'sshd:x:74:' ) >> "$initdir/etc/group"
```

Comment:
What is the use-case for the `echo 'sshd: ...` part? I can't imagine a system where the `getents` would fail.

If possible I would just use the `getent` part then.

Reply:
Special case for Silverblue (and I assume rpm-ostree at all). The whole Dracut install thing happens in a chroot where sssd is not available and neither `/etc/passwd` nor `/etc/group` contain sshd. I only could come up with this workaround to make it work, but at least it works on both my Silverblue 33 systems this way.

Reply:
But why include the getent part then?